src/duplicacy_b2client.go

```diff
@@ -94,19 +94,22 @@ func (client *B2Client) call(url string, input interface{}) (io.ReadCloser, int6
 		default:
 			jsonInput, err := json.Marshal(input)
 			if err != nil {
-				return nil, 0, err
+				return nil, nil, 0, err
 			}
 			inputReader = bytes.NewReader(jsonInput)
 		case []byte:
 			inputReader = bytes.NewReader(input.([]byte))
 		case int:
 			method = "GET"
 			inputReader = bytes.NewReader([]byte(""))
+		case bool:
```

Comment:
You're right, there are the constants in the http package. Those should be used here. https://golang.org/pkg/net/http/#pkg-constants

More generally, I think this can be cleaner by passing in a map of headers, input as a bytes.Reader, and the method as string (a switch can restrict this to only match the http constants)

```
call(url string, requestMethod string, requestHeaders map[string]string, input *bytes.Reader)

…

switch requestMethod {
    default:
        return nil, nil, 0, err
    case MethodGet:
        break
    case MethodHead:
        break
    case MethodPost:
        break
}
```

Reply:
What you've proposed for the request method sounds fine given that strongly-typed enums don't exist in Go.

I'm not wild about the suggestion that creating the bytes.Reader should be done each place that the method is called; that feels like it'll lead to code duplication. Rather, keep the current parameter type of interface, update the logic to allow passing nil for the cases where empty content should be used for the request, and remove the special cases for int and bool (passing nil instead).

Reply:
I started thinking about the duplication as well. One idea was making use of lambdas. I haven't had a chance to actually implement anything yet and lambdas seem heavy handed here as well. I'll try your suggestion.